matrix/matrix_class.py

```diff
@@ -286,7 +286,7 @@ def add_column(self, column: list[int], position: int | None = None) -> None:
     # MATRIX OPERATIONS
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, Matrix):
-            raise TypeError("A Matrix can only be compared with another Matrix")
+            return False

```

Comment:
Actually, it should return `NotImplemented`
```suggestion
            return NotImplemented
```

Reply:
Really? Okay.

Reply:
Yes, in dunder methods for operators, if you return `NotImplemented`, Python will automatically raise an error with an appropriate error message. You can try it out with some `Foo` class.